sqlglot/dialects/prql.py

```diff
@@ -136,6 +153,25 @@ def _parse_order_by(self, query: exp.Select) -> t.Optional[exp.Query]:
                 self.raise_error("Expecting }")
             return query.order_by(self.expression(exp.Order, expressions=expressions), copy=False)
 
+        def _parse_aggregate(self) -> t.Optional[exp.Expression]:
+            alias = None
+            if self._next and self._next.token_type == TokenType.ALIAS:
+                alias = self._parse_id_var(True)
+                self._match(TokenType.ALIAS)
+
+            name = self._curr and self._curr.text.upper()
+            func_builder = self.FUNCTIONS.get(name)
+            if func_builder:
+                self._advance()
+                # "this" keyword in PRQL refers to the current relation
+                args = exp.Star() if self._match_texts("THIS") else self._parse_id_var()
```

Comment:
This is not tested and it doesn't look like it'll suffice to handle `this`. You have to know what the "current" relation is so that you can query it in the produced SQL.

We'll probably need additional context to achieve this because we've already parsed the name of said relation before reaching this method, so perhaps we could leave `THIS` as is and replace it later, after the parsing is complete, as a post-transformation of the AST or something.

In any case, I'd leave `THIS` out of this one and tackle it in another PR later.

Reply:
I don't think `_parse_id_var()` is correct btw, you want `_parse_column` - from an AST perspective, a column is not an identifier.